src/bulk-transfer-mapping.ts

```diff
@@ -0,0 +1,71 @@
+import { log } from '@graphprotocol/graph-ts'
+
+import {
+  BatchPrepared as BatchPreparedEvent,
+  ApproveSRRByCommitmentWithProof as ApproveSRRByCommitmentWithProofEvent,
+  MigrateBatch as MigrateBatchEvent,
+} from '../generated/BulkTransfer/BulkTransfer'
+import { BulkTransfer } from '../generated/schema'
+import { eventUTCMillis, logInvocation, secondsToMillis } from './utils'
+
+export function handleBatchPrepared(event: BatchPreparedEvent): void {
+  logInvocation("handleBatchPrepared", event);
+
+  let merkleRoot = event.params.merkleRoot.toHexString();
+  let batch = BulkTransfer.load(merkleRoot);
+  if (batch != null) {
+    log.info("already received this event for merkleRoot: {}", [
+      event.params.merkleRoot.toString(),
+    ]);
+    return;
+  }
+
+  batch = new BulkTransfer(merkleRoot);
+  batch.txs = [];
+  batch.merkleRoot = event.params.merkleRoot;
+  batch.sender = event.params.sender;
+
+  batch.createdAt = batch.updatedAt = eventUTCMillis(event);
+  batch.save();
+}
+
+export function handleCreateSRRWithProof(event: ApproveSRRByCommitmentWithProofEvent): void {
```

Comment:
createSRR in BulkTransfer?  this is handleApprove... right?

Reply:
update name used in the logging below too. logInvocation and log.error ...

Reply:
Sorry, it. is a copy from bulk, fixed it.